test/per_file_coverage.sh

```diff
@@ -30,8 +30,8 @@ declare -a KNOWN_LOW_COVERAGE=(
 "source/common/upstream:96.0"
 "source/common/watchdog:58.6" # Death tests don't report LCOV
 "source/exe:92.6"
-"source/extensions/common:95.8"
-"source/extensions/common/tap:92.9"
+"source/extensions/common:95.7"
+"source/extensions/common/tap:91.5"
```

Comment:
can you add tests in this PR rather than slackening our coverage expectations?

Reply:
The reason why I reduced the coverage bounds is because the uncovered code lies in exceptional cases & race cases. I probably should have added a blurb in the initial PR about this - my bad on that. 

Are there any examples I could follow of how race cases & exceptional cases are tested in Envoy to improve coverage here? I was following the tests for `tap_config_base.cc` but they don't seem to cover the exceptional cases either. For example in `TapConfigBaseImpl::TapConfigBaseImpl` the cases in the `switch` statement causing `PANIC`s are not covered.

Here is the [link](https://dev.azure.com/cncf/4684fb3d-0389-4e0b-8251-221942316e06/_apis/build/builds/102861/artifacts?artifactName=bazel.coverage&api-version=7.0&%24format=zip) to the coverage output for my changes - let me know if there are particular areas you think should be covered better!

Reply:
Let me take a look at this. the PANICs in switch statements are annoying for sure, and we used to have a better way to exclude lines from coverage.

For reproducing race conditions PTAL @ https://github.com/envoyproxy/envoy/blob/main/source/common/common/thread_synchronizer.h

Reply:
For covering trivial overrides I think it's fine just to call them, a la https://github.com/envoyproxy/envoy/blob/be8b2594cb5d157e36f39b996a7715812d2e68ae/test/server/admin/admin_test.cc#L253